lib/connection/result/column.js

```diff
@@ -268,15 +284,277 @@ function convertRawBigInt(rawColumnValue) {
+      result.set(convertedKey, convertedValue);
+    });
+
+    return result;
+  }
+}
+
+/**
+ * Converts a raw column value of structured type OBJECT to javascript Object
+ *
+ * @param {String} rawColumnValue
+ * @param {Object} column
+ * @param {Object} context
+ *
+ * @returns {Object}
+ */
+function convertRawObject(rawColumnValue, column, context) {
+  if (Util.string.isNotNullOrEmpty(rawColumnValue)) {
+    try {
+      const json = JSON.parse(rawColumnValue);
+      return convertJsonObject(json, context);
+    } catch (jsonParseError) {
+      Logger.getInstance().debug('Column %s raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message);
+      throw new Error(util.format('Column [%s] raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message));
+    }
+  } else {
+    throw new Error(util.format('Column %s raw value is null or empty ', column.name));
+  }
+}
+/**
+ * Converts a raw column value of structured type ARRAY to javascript Array
+ *
+ * @param {String} rawColumnValue
+ * @param {Object} column
+ * @param {Object} context
+ *
+ * @returns {Object}
+ */
+function convertRawArray(rawColumnValue, column, context) {
+  if (Util.string.isNotNullOrEmpty(rawColumnValue)) {
+    try {
+      const json = JSON.parse(rawColumnValue);
+      return convertJsonArray(json, context);
+    } catch (jsonParseError) {
+      Logger.getInstance().debug('Column %s raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message);
+      throw new Error(util.format('Column [%s] raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message));
+    }
+  } else {
+    throw new Error(util.format('Column %s raw value is null or empty ', column.name));
+  }
+}
+/**
+ * Converts a raw column value of structured type MAP to javascript Map
+ * @param {String} rawColumnValue
+ * @param {Object} column
+ * @param {Object} context
+ *
+ * @returns {Object}
+ */
+function convertRawMap(rawColumnValue, column, context) {
+  if (Util.string.isNotNullOrEmpty(rawColumnValue)) {
+    try {
+      const json = JSON.parse(rawColumnValue);
+      return convertJsonMap(json, context);
+    } catch (jsonParseError) {
+      Logger.getInstance().debug('Column %s raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message);
+      throw new Error(util.format('Column [%s] raw value cannot be parsed as JSON: %s ', column.name, jsonParseError.message));
+    }
+  } else {
+    throw new Error(util.format('Column %s raw value is null or empty ', column.name));
+  }
+}
+
+function mapStructuredTypeValue(columnValue, context, metadataField) {
+  const formatLtz = context.statementParameters['TIMESTAMP_LTZ_OUTPUT_FORMAT'] || context.statementParameters['TIMESTAMP_OUTPUT_FORMAT'];
+  const formatTz = context.statementParameters['TIMESTAMP_TZ_OUTPUT_FORMAT'] || context.statementParameters['TIMESTAMP_OUTPUT_FORMAT'];
```

Comment:
I'd consider using ?? instead of || to avoid confusion with logical or

Reply:
Ok. Fixed.